package com.harbe.orderservice.exception;


import com.harbe.orderservice.dto.exception.ErrorDetails;
import com.harbe.orderservice.exception.HarbeAPIException;
import com.harbe.orderservice.exception.ResourceNotFoundException;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpStatus;
import org.springframework.http.HttpStatusCode;
import org.springframework.http.ResponseEntity;
import org.springframework.security.access.AccessDeniedException;
import org.springframework.validation.FieldError;
import org.springframework.web.bind.MethodArgumentNotValidException;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.context.request.WebRequest;
import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;

import java.util.Date;
import java.util.HashMap;
import java.util.Map;

public class OrderExceptionHandler extends ResponseEntityExceptionHandler{

    // handle specific exceptions
    @ExceptionHandler(Exception.class)
    public ResponseEntity<ErrorDetails> harbeOrderExcpetionHanlder(Exception exception,
                                                                   WebRequest webRequest){
        ErrorDetails errorDetails = new ErrorDetails(new Date(), exception.getMessage(),
                webRequest.getDescription(false));
        return new ResponseEntity<>(errorDetails, HttpStatus.INTERNAL_SERVER_ERROR);
    }

}

